adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,32 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Converts this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage(collapse: Boolean = true): CoverageRDD = {
+    val coverage: RDD[Coverage] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
+        .sortByKey()
+        .map(r => Coverage(r._1, r._2.toDouble))
+
+    val coverageRdd =
+      if (collapse)
+        CoverageRDD(coverage, sequences).collapse
+      else CoverageRDD(coverage, sequences)
+
+    coverage.unpersist()
```

Comment:
You don't want to unpersist `coverage`. You'd want to unpersist `covCounts` from https://github.com/bigdatagenomics/adam/pull/1108/files#r75248157. Calling unpersist on `coverage` should cause everything done to the `CoverageRDD` returned by this method to throw an exception...